File: guests/ocaml/grid.ml
(* The grid conformance scene, OCaml port. See
   guests/rust/grid.rs and tools/scenes/grid.steps. *)

open Kaya_app

let () =
  let app = Kaya_app.create () in

  build app (fun () ->
     window ~title:"grid" ();
     let root =
       column
         [
           grid ~columns:2
             [
               label ~text:"Name:" (* label#0 *);
               label ~text:"Ada Lovelace" (* label#1 *);
               label ~text:"Role:" (* label#2 *);
               label ~text:"Engine programmer" (* label#3 *);
             ];
           row ~grow:1.0
             [
               button ~text:"left" (* button#0 *);
               spacer ~grow:1.0;
               button ~text:"right" (* button#1 *);
             ];
         ]
         ()
     in
     mount root);

  exit (run app)

File: guests/swift/grid.swift
// The grid scene, Swift port — guests/rust/grid.rs, tools/scenes/grid.steps.

import Foundation
import Kaya

KayaApp.run { app in
    app.build { tx in
        tx.window(title: "grid")
        let root = tx.column { root in
            tx.grid(columns: 2) { _ in
                tx.label("Name:")  // label#0
                tx.label("Ada Lovelace")  // label#1
                tx.label("Role:")  // label#2
                tx.label("Engine programmer")  // label#3
            }
            tx.row(grow: 1.0) { _ in
                tx.button("left")  // button#0
                tx.spacer()
                tx.button("right")  // button#1
            }
            return root
        }
        tx.mount(root)
    }
}
